switching/input/messages/Q1.py

```diff
@@ -188,6 +188,50 @@ def get_lectures(self):
             pass
         return lectures
 
+    def get_lectures_activa(self):
+        lectures_activa = []
+        try:
+            for lect in self.obj.Integrador:
+                lectura = Lectura(lect)
+                if lectura.tipus == 'A':
+                    lectures_activa.append(lectura)
+        except AttributeError:
+            pass
+        return lectures_activa
+
+    def get_lectures_reactiva(self):
+        lectures_reactiva = []
+        try:
+            for lect in self.obj.Integrador:
+                lectura = Lectura(lect)
+                if lectura.tipus == 'R':
+                    lectures_reactiva.append(lectura)
+        except AttributeError:
+            pass
+        return lectures_reactiva
+
+    def get_lectures_energia(self):
+        lectures_energia = []
+        try:
+            for lect in self.obj.Integrador:
+                lectura = Lectura(lect)
+                if lectura.tipus in ('A', 'R'):
+                    lectures_energia.append(lectura)
+        except AttributeError:
+            pass
+        return lectures_energia
+
+    def get_lectures_maximetre(self):
+        lectures_maximetre = []
+        try:
+            for lect in self.obj.Integrador:
+                lectura = Lectura(lect)
+                if lectura.tipus == 'M':
+                    lectures_maximetre.append(lectura)
+        except AttributeError:
+            pass
+        return lectures_maximetre
```

Comment:
One Idea to avoid to duplicate code could be:
```python
def get_lectures(*tipus):
    lectures = []
    try:
       for lect in self.obj.Integrador:
           lectura = Lectura(lect)
           if lectura.tipus in tipus:
               lectures.append(lectura)
    except AttributeError:
        pass
    return lectures

def get_lectures_maximetre(self):
    return self.get_lectures('M')

def get_lectures_energia(self):
    return self.get_lectures('A', 'R')
```